### PR TITLE
Properties

### LINKED ISSUE #957: Get property history without times
The `property_history` function currently returns a list of tuples containing the value and the time the value was set. This is useful, but sometimes the times are unneeded, meaning you have to map over the list to extract only the value. This can make code unnecessarily ugly and should be internalised. 

There are several ways we could handle this, but the simplest could be a flag:

```python 
print(graph.vertex(1).property_history("prop 1"))
print(graph.vertex(1).property_history("prop 1", include_time=False))
```

```python
[(1,2),(5,7),(14,22)]
[2,7,22]
```

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

Unify the properties api and create Properties and StaticProperties objects

### Why are the changes needed?

Make working with properties easier to use and properties api easier to implement

### Does this PR introduce any user-facing change? If yes is this documented?

yes, all the properties apis are different. Rudimentary docstrings are there but need to be improved

### Issues

closes #957, #956

### CODE REVIEW

**Review by miratepuffin — CHANGES_REQUESTED** [review]
A couple of small points, mainly on documentation:

- Seems we have lost a lot of python docs
- We need to manually add the properties object to the docs rust

File: raphtory/src/algorithms/triangle_count.rs [hunk 1]
```diff
@@ -67,6 +67,7 @@ pub fn triangle_count<G: GraphViewOps>(g: &G, threads: Option<usize>) -> usize {
     let step1 = ATask::new(move |s| {
         for t in s.neighbours() {
             if s.id() > t.id() {
+                println!("adding {} to neighbours of {}", s.id(), t.id());
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/algorithms/triangle_count.rs:70]: Yes, this was from testing the subgraph bug, should not be here obviously...

File: raphtory/src/db/api/properties/props.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            .and_then(|k| self.props.temporal_value(&k))
+            .or_else(|| self.props.get_const_property(key.as_ref()))
+    }
+
+    /// Check if property `key` exists.
+    pub fn contains<Q: AsRef<str>>(&self, key: Q) -> bool {
+        self.get(key).is_some()
+    }
+
+    pub fn keys(&self) -> impl Iterator<Item = LockedView<String>> + '_ {
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/api/properties/props.rs:36]: yes

File: raphtory/src/db/graph/vertices.rs [hunk 3]
```diff
@@ -72,116 +74,12 @@ impl<G: GraphViewOps> VertexViewOps for Vertices<G> {
...
-    }
-
-    /// Returns static properties of the vertices
-    ///
-    /// # Returns
-    ///
-    /// An iterator of the static properties of the vertices
-    fn static_properties(&self) -> Self::ValueType<HashMap<String, Prop>> {
-        self.iter().static_properties()
+    fn properties(&self) -> Self::ValueType<Properties<VertexView<G>>> {
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/graph/vertices.rs:81]: should indeed have it as it is different from the default docstring of the trait

File: raphtory/src/db/api/view/vertex.rs [hunk 4]
```diff
@@ -223,56 +142,10 @@ pub trait VertexListOps:
...
-    /// including the times when it changed
-    ///
-    /// # Arguments
-    /// * `name` - The name of the property.
-    ///
-    /// # Returns
-    /// An iterator of the values of the given property name including the times when it changed
-    /// as a vector of tuples of the form (time, property).
-    fn property_history(self, name: String) -> Self::IterType<Vec<(i64, Prop)>>;
+    /// Returns an iterator over properties of the vertices
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/api/view/vertex.rs:145]: Agreed, we should do that when we refactor the internal implementations

**Comment by ljeub-pometry** [comment]
I cleaned up the minor comments and added python docs for the base types. Still need to figure out the best way to document the lists and lists of lists variants

**Review by miratepuffin — APPROVED** [review]

**Pull request merged**